core/services/keeper/registry_interface.go

```diff
@@ -119,8 +125,39 @@ func (rw *RegistryWrapper) GetActiveUpkeepIDs(opts *bind.CallOpts) ([]*big.Int,
 		}
 		return activeUpkeeps, nil
 	case RegistryVersion_1_2:
-		// TODO (sc-37024): Get active upkeep IDs from contract in batches
-		return rw.contract1_2.GetActiveUpkeepIDs(opts, big.NewInt(0), big.NewInt(0))
+		// fetch the current block number so batched GetActiveUpkeepIDs calls can be performed on the same block
+		header, err := rw.evmClient.HeaderByNumber(context.Background(), (*big.Int)(nil))
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to fetch EVM block header")
+		}
+		if opts != nil {
+			opts.BlockNumber = header.Number
+		} else {
+			opts = &bind.CallOpts{
+				BlockNumber: header.Number,
+			}
+		}
+
+		state, err := rw.contract1_2.GetState(opts)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to get contract state")
+		}
+
+		activeUpkeepIDs := make([]*big.Int, 0)
+		for int64(len(activeUpkeepIDs)) < state.State.NumUpkeeps.Int64() {
+			startIndex := int64(len(activeUpkeepIDs))
+			activeUpkeepIDBatch, err := rw.contract1_2.GetActiveUpkeepIDs(opts, big.NewInt(startIndex), big.NewInt(ActiveUpkeepIDBatchSize))
+			if err != nil {
+				return nil, errors.Wrapf(err, "failed to get active upkeep IDs from index %v to %v", startIndex, startIndex+ActiveUpkeepIDBatchSize)
```

Comment:
nit. I normally use %d to print int but [there isn't much difference](https://stackoverflow.com/questions/41159169/why-not-use-v-to-print-int-and-string).